components/provisioner/internal/model/gardener_config.go

```diff
@@ -478,6 +478,9 @@ func (c AzureGardenerConfig) EditShootConfig(gardenerConfig GardenerConfig, shoo
 
 		if len(c.input.AzureZones) == 0 {
 			if *c.input.EnableNatGateway {
+				if infra.Networks.NatGateway == nil {
+					infra.Networks.NatGateway = &azure.NatGateway{}
+				}
```

Comment:
When looking at the code I can see very similar problem few lines below:

When setting NatGateway for each zone we can have a panic when referencing: 
`zone.NatGateway.Enabled` without creating it first.

So suggested fix would be :

```
for i := range infra.Networks.Zones {
				zone := infra.Networks.Zones[i]
				if *c.input.EnableNatGateway {
                                          if zone.NatGateway == nil {
                                             zone.NatGateway = &azure.NatGateway{}
                                          } 
					zone.NatGateway.Enabled = *c.input.EnableNatGateway
					zone.NatGateway.IdleConnectionTimeoutMinutes = util.UnwrapIntOrDefault(c.input.IdleConnectionTimeoutMinutes, defaultConnectionTimeOutMinutes)
				} else {
					zone.NatGateway = nil
				}
				infra.Networks.Zones[i] = zone
```

Or we can just revert [all changes made here](https://github.com/kyma-project/control-plane/pull/2920/files#diff-503774e2175c9715d5639e33b7b8cd1a66097f93490cd49b6ed4d4ea9e5c1775R478-R492) for PR-2920